<launch>

    <arg name="sonars_installed" default="true"/>

    <!-- Run pi_sonar for obstacle detection -->
    <include file="$(find pi_sonar)/launch/ubiquity_sonar.launch" if="$(arg sonars_installed)" />

    <!-- Run move_smooth for navigation -->
    <node name="move_smooth" pkg="move_smooth" type="move_smooth" output="screen">
	 <!-- Footprint for obstacle detection in collision checker-->
	 <param name="robot_width" value="0.20"/>
	 <param name="robot_front_length" value="0.1"/>
	 <param name="robot_back_length" value="0.32"/>

	 <param name="max_angular_velocity" value="2.0"/>
	 <param name="angular_acceleration" value="0.3"/>
	 <param name="max_linear_velocity" value="0.4"/>
	 <param name="linear_acceleration" value="0.3"/>
	 <param name="linear_tolerance" value="0.7"/>
	 <param name="intermediate_linear_tolerance" value="0.7"/>

	 <param name="lateral_kp" value="0.2"/>
	 <param name="lateral_ki" value="0.0"/>
	 <param name="lateral_kd" value="8.0"/>

	 <param name="max_incline_without_slipping" value="0.01"/>
	 <param name="gravity_acceleration" value="9.81"/>
	 <param name="max_lateral_deviation" value="0.4"/>

	 <param name="min_side_dist" value="0.3"/>
	 <param name="minimum_requested_distance" value="0.2"/>
	 <param name="side_recover_weight" value="0.3"/>

	 <param name="forward_obstacle_threshold" value="0.7"/>

	 <param name="abort_timeout" value="60.0"/>

	 <param name="preferred_driving_frame" value="map"/>
	 <param name="alternate_driving_frame" value="odom"/>
	 <param name="base_frame" value="base_link"/>
    </node>

</launch>
